lib/trellis/plugins/vars/vars.py

```diff
@@ -20,14 +23,57 @@ def __init__(self, inventory):
         self.inventory_basedir = inventory.basedir()
         self._options = cli.options if cli else None
 
-    # Wrap salts and keys variables in {% raw %} to prevent jinja templating errors
-    def wrap_salts_in_raw(self, host, hostvars):
-        if 'vault_wordpress_sites' in hostvars:
-            for name, site in hostvars['vault_wordpress_sites'].iteritems():
-                for key, value in site['env'].iteritems():
-                    if key.endswith(('_key', '_salt')) and not value.startswith(('{% raw', '{%raw')):
-                        hostvars['vault_wordpress_sites'][name]['env'][key] = ''.join(['{% raw %}', value, '{% endraw %}'])
-            host.vars['vault_wordpress_sites'] = hostvars['vault_wordpress_sites']
+    def raw_triage(self, key_string, item, patterns):
+        # process dict values
+        if isinstance(item, AnsibleMapping):
+            dict = {}
+            for key,value in item.iteritems():
+                dict[key] = self.raw_triage('.'.join([key_string, key]), value, patterns)
+            return dict
+
+        # process list values
+        elif isinstance(item, AnsibleSequence):
+            list = []
+            for i,value in enumerate(item):
+                raw = self.raw_triage('.'.join([key_string, str(i)]), value, patterns)
+                list.append(raw)
+            return list
+
+        # wrap values if they match raw_vars pattern
+        elif isinstance(item, AnsibleUnicode):
+            matches = False
+            for pattern in patterns:
+                if re.match(pattern, key_string) is not None:
+                    matches = True
+                    break
+
+            if not item.startswith(('{% raw', '{%raw')) and matches:
+                item = ''.join(['{% raw %}', item, '{% endraw %}'])
+
+            return item
+
+    def raw_vars(self, host, hostvars):
+        if 'raw_vars' not in hostvars:
+            return
```

Comment:
That "visibility" was my focus. My overarching goal was to transition this functionality from "hidden" -- which for users is possibly surprising and certainly non-configurable -- to "out in the open" and easy to use.

I agree most users won't understand it and won't touch it, in which case they won't mess it up. In that sense, I think it is like neighboring obscure vars in `group_vars/all/main.yml`, such as `wordpress_env_defaults` or `site_env`. 

For users who do try to use it, actually adding patterns versus merely disabling it, I think they'll have the easiest time if they readily see the full list of `raw_vars` default patterns right there in a group_vars file.

Would users really want to adjust patterns? I think a conceivable example would be that someone wants to add an env var in `vault_wordpress_sites` that should _not_ be wrapped in `{% raw %}`. They'd need to adjust the `raw_vars` to be more specific, just wrapping the right parts, such as:
```
raw_vars:
  - vault_wordpress_sites.*.*_password       # for `admin_password` (dev only) and `db_password`
  - vault_wordpress_sites.*.*_salt
  - vault_wordpress_sites.*.*_key
```

On the other hand, if we decide that visibility and configurability aren't the goal or aren't worth it, then I suppose moving the defaults into `vars.py` makes sense. In that case, we could consider whether the python code would be "safer" being more targeted to those defaults, not supporting the wildcard stuff and being more "hard-coded" like it has been for WP keys and salts.